### PR TITLE
feat: cypress test coverage reporting

### PR DESCRIPTION
**Pull request by just-toby**
adds instrumentation to the app for test runs, and enables coverage reporting in cypress.

pros:
- provides an ongoing guidance about where to focus our attention for new e2e tests (which are be more powerful than unit tests)
- more accurate reporting of exactly how much code is actually executing automatically in the test runs
- number go up

cons:
- e2e tests might execute a line of code, but they won't actually test functionality in specific ways like unit tests would. the only test is the high-level user flow. we might not want to consider these lines "covered" 
- performance of e2e test runs could be slightly worse (i haven't tested this yet) since the build has a bunch of instrumentation code now.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -178,6 +179,7 @@
     "@web3-react/walletconnect": "8.1.2-beta.0",
     "array.prototype.flat": "^1.2.4",
     "array.prototype.flatmap": "^1.2.4",
+    "babel-plugin-istanbul": "^6.1.1",
```
Comment (JFrankfurt) [inline package.json:182]: what's up with this babel plugin?

Comment (just-toby) [reply package.json:182]: this is what actually does the instrumentation: https://docs.cypress.io/guides/tooling/code-coverage#Using-code-transpilation-pipeline

**Review by cmcewen — APPROVED** [review]
have you tested to see if this has any speed impact due to collecting code coverage? not sure how it works in e2es tbh

**Comment by just-toby** [comment]
> have you tested to see if this has any speed impact due to collecting code coverage? not sure how it works in e2es tbh

the Github actions are not taking significantly longer, so the speed impact seems low

**Pull request merged**